pkg/provisioner/controller.go

```diff
@@ -380,7 +470,43 @@ func (c *obcController) handleProvisionClaim(key string, obc *v1alpha1.ObjectBuc
 		return fmt.Errorf("error getting reference to OBC: %v", err)
 	}
 
-	// create Secret and ConfigMap
+	// Create auth Secret and bucket info ConfigMap
+	// If the operator crashes before the OB is created, provisioning will happen again including a
+	// call to Provision()/Grant(), and a Secret and/or ConfigMap might exist from the previous
+	// provisioning attempt. Provision() isn't guaranteed to give the same credentials or bucket
+	// each time, so if the Secret/ConfigMap already exists, delete the old one and re-create it
+	// with the latest info.
+
+	secret, err = getExistingSecret(obc, c.clientset, defaultRetryBaseInterval, defaultRetryTimeout)
+	if err != nil {
+		return fmt.Errorf("error checking for existing Secret for OBC: %v", err)
+	}
+	if secret != nil {
+		if !isOwnedByClaim(obc, secret.OwnerReferences) {
+			return fmt.Errorf("found an existing OBC credentials Secret that is not owned by this OBC, this may be an artifact from a previous OBC that did not get cleaned up or a user-created resource, not continuing with OBC provisioning")
```

Comment:
Isn't it possible that the secret was created but the ownership part, which is set in a separate step, is where we crashed? In this case the secret is correct but only partially filled in. Since secrets are immutable we cannot edit it so I agree with erroring out. I'd like to see the error msg suggest to delete the secret.

Reply:
If we error out, we cause infinite provisioning loops, which is what was the root cause of the infinite provision loop fixed in #198 and can cause cleanup and data loss now that #198 is in. The secret needs to be able to be updated so this cannot happen.

Reply:
Maybe I'm missing something here. 1) When I wrote "error out" I was referring to L486 and L491 so I don't understand your reply of "_If we error out, we cause infinite provisioning loops_" because we are error-ing out (at least by me definition :-). 2) You wrote "_The secret needs to be able to be updated_", but Secrets are immutable and the code handles this by deleting the Secret and creating a new one. 
The interesting case here is when the secret name matches what the lib wants to create but the ownership doesn't match, and in that case I was suggesting the error msg suggest that this unexpected secret be deleted.

Reply:
Okay. I thought there was more intent other than suggesting users delete the secret (or configmap in that case).